### PR TITLE
chore(link): fix url

### LINKED ISSUE #979: bug(chats): Message with a text starting with "www." is sent as link but the link is not working
1. Go to a chat
2. Type in the chat input bar a link starting with www. (like www.apple.com) and send message
3. Message sent is formatted as link without a preview
4. Click on the link just sent and we will see that the link is not working

Note: This is only happening with links sent starting with www.
If you send the link starting with https:// or http:// the link redirects to the URL

image

### PR DESCRIPTION
**Pull request by stavares843**
### What this PR does 📖

When you send www.google.pt in the chat wasn't sent as a URL and wasn't clickable 

now is a hyperlink and opens the website

image

### CODE REVIEW

**Review by phillsatellite — APPROVED** [review]

**Comment by WanderingHogan** [comment]
www.google.com
gist.github.com
news.ycombinator.com

**Comment by WanderingHogan** [comment]
at first i thought this was silly, why treat www special over other subdomains, but i can see this being useful to regular users, so i like it

File: kit/src/components/message/mod.rs [hunk 1]
```diff
@@ -88,11 +87,15 @@ pub struct Props<'a> {
...
 
 fn wrap_links_with_a_tags(text: &str) -> String {
-    let re = Regex::new(r#"(?i)\b((?:https?://|www\d{0,3}[.]|[a-z0-9.\-]+[.][a-z]{2,4}/)(?:[^\s()<>]+|\(([^\s()<>]+|(\([^\s()<>]+\)))*\))+(?:\(([^\s()<>]+|(\([^\s()<>]+\)))*\)|[^\s`!()\[\]{};:'".,<>?«»“”‘’]))"#).unwrap();
+    let re = regex::Regex::new(r#"(?i)\b((?:(?:https?://|www\.)[^\s()<>]+|\(([^\s()<>]+|(\([^\s()<>]+\)))*\))+(?:\(([^\s()<>]+|(\([^\s()<>]+\)))*\)|[^\s`!()\[\]{};:'".,<>?«»“”‘’]))"#).unwrap();
 
     re.replace_all(text, |caps: &regex::Captures| {
         let url = caps.get(0).unwrap().as_str();
-        format!("<a href=\"{}\">{}</a>", url, url)
+        if url.starts_with("www.") {
+            format!("<a href=\"http://{}\">{}</a>", url, url)
```
Comment (WanderingHogan) [inline kit/src/components/message/mod.rs]: we should force https here and not rely on forced redirects on whatever site they are on

**Review by WanderingHogan — APPROVED** [review]

File: kit/src/components/message/mod.rs (hunk 1 above)
Comment (stavares843) [inline kit/src/components/message/mod.rs]:
```suggestion
            format!("<a href=\"https://{}\">{}</a>", url, url)
```

Comment (stavares843) [reply kit/src/components/message/mod.rs]: done 🎉

**Comment by luisecm** [comment]
I like it too!, now typing with http://,  https:// or www. should work in the same way

**Comment by stavares843** [comment]
🎉

File: kit/src/components/message/mod.rs [hunk 2]
```diff
@@ -88,11 +87,15 @@ pub struct Props<'a> {
...
 
 fn wrap_links_with_a_tags(text: &str) -> String {
-    let re = Regex::new(r#"(?i)\b((?:https?://|www\d{0,3}[.]|[a-z0-9.\-]+[.][a-z]{2,4}/)(?:[^\s()<>]+|\(([^\s()<>]+|(\([^\s()<>]+\)))*\))+(?:\(([^\s()<>]+|(\([^\s()<>]+\)))*\)|[^\s`!()\[\]{};:'".,<>?«»“”‘’]))"#).unwrap();
+    let re = regex::Regex::new(r#"(?i)\b((?:(?:https?://|www\.)[^\s()<>]+|\(([^\s()<>]+|(\([^\s()<>]+\)))*\))+(?:\(([^\s()<>]+|(\([^\s()<>]+\)))*\)|[^\s`!()\[\]{};:'".,<>?«»“”‘’]))"#).unwrap();
 
     re.replace_all(text, |caps: &regex::Captures| {
         let url = caps.get(0).unwrap().as_str();
-        format!("<a href=\"{}\">{}</a>", url, url)
+        if url.starts_with("www.") {
+            format!("<a href=\"https://{}\">{}</a>", url, url)
```
Comment (dariusc93) [inline kit/src/components/message/mod.rs:95]: Just for clarification, are we to assume every url will be https? Not that this is blocking or anything but just curious (although it is clear from the regex that it is)

**Review by dariusc93 — APPROVED** [review]

**Review by lgmarchi — APPROVED** [review]

File: kit/src/components/message/mod.rs (hunk 2 above)
Comment (stavares843) [reply kit/src/components/message/mod.rs:95]: initially i added http but hogan suggested to force the https instead 🔨  what do you think we should improve? 💥

**Pull request merged**